airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -127,17 +206,34 @@ def state_checkpoint_interval(self) -> int:
 
     # Setting the default cursor field for all streams
     cursor_field = "updated_at"
+    deleted_cursor_field = "deleted_at"
+
+    @property
+    def deleted_events_instance(self) -> ShopifyDeletedEventsStream:
+        return ShopifyDeletedEventsStream(self.config, self.data_field, self.events_name)
 
     @property
     def default_state_comparison_value(self) -> Union[int, str]:
         # certain streams are using `id` field as `cursor_field`, which requires to use `int` type,
         # but many other use `str` values for this, we determine what to use based on `cursor_field` value
         return 0 if self.cursor_field == "id" else ""
 
+    @property
+    def default_deleted_state_comparison_value(self) -> Union[int, str]:
+        # `deleted_at` is always a type `str`
+        return ""
+
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
         last_record_value = latest_record.get(self.cursor_field) or self.default_state_comparison_value
         current_state_value = current_stream_state.get(self.cursor_field) or self.default_state_comparison_value
-        return {self.cursor_field: max(last_record_value, current_state_value)}
+        state = {self.cursor_field: max(last_record_value, current_state_value)}
```

Comment:
conceptually speaking, updating the state of retrieving deleted events should not impact the state of the stream itself. Can we keep these things separate? i.e if we're retrieving deleted records that should not update the "normal" stream state

Reply:
The  state is stored like:
```json
{
        "type": "STREAM",
        "stream": {
            "stream_state": {
                "deleted": {
                    "deleted_at": <deleted_state_value>
                },
                "updated_at": <stream_cursor_field_state_value>
            },
            "stream_descriptor": {
                "name": "products"
            }
        }
    },
```

The removed records are being updated separately from stream records,  thus the state value for them related to the stream is stored within  `the "deleted"` property inside the state, they are not mixed together and used when it's needed.
The stream's STATE remains intact here, we just extend the STATE with another property we could use later. They could be treated separately only if we make 7 new streams for removed records specifically (which doesn't make sense to me). Please confirm you want 7 new streams for removed records so I can be sure we are on the same page.

Reply:
I've moved the logic to a separate class, making the original Incremental ShopifyStream class intact by this change. Thus we now guarantee the presence of the  `deleted_events` instance for those streams where it's used, the other ones remain as they are.